firmware/targets/f7/furi_hal/furi_hal_nfc_iso15693.c

```diff
@@ -0,0 +1,463 @@
+#include "furi_hal_nfc_i.h"
+#include "furi_hal_nfc_tech_i.h"
+
+#include <digital_signal/presets/nfc/iso15693_signal.h>
+#include <signal_reader/parsers/iso15693/iso15693_parser.h>
+
+#include <furi_hal_resources.h>
+
+#define FURI_HAL_NFC_ISO15693_MAX_FRAME_SIZE (1024U)
+#define FURI_HAL_NFC_ISO15693_POLLER_MAX_BUFFER_SIZE (64)
+
+#define FURI_HAL_NFC_ISO15693_RESP_SOF_SIZE (5)
+#define FURI_HAL_NFC_ISO15693_RESP_EOF_SIZE (5)
+#define FURI_HAL_NFC_ISO15693_RESP_SOF_MASK (0x1FU)
+#define FURI_HAL_NFC_ISO15693_RESP_SOF_PATTERN (0x17U)
+#define FURI_HAL_NFC_ISO15693_RESP_EOF_PATTERN (0x1DU)
+
+#define FURI_HAL_NFC_ISO15693_RESP_PATTERN_MASK (0x03U)
+#define FURI_HAL_NFC_ISO15693_RESP_PATTERN_0 (0x01U)
+#define FURI_HAL_NFC_ISO15693_RESP_PATTERN_1 (0x02U)
+
+// Derived experimentally
+#define FURI_HAL_NFC_ISO15693_POLLER_FWT_COMP_FC (-1300)
+#define FURI_HAL_NFC_ISO15693_LISTENER_FDT_COMP_FC (2850)
+
+#define BITS_IN_BYTE (8U)
+
+#define TAG "FuriHalIso15693"
+
+typedef struct {
+    Iso15693Signal* signal;
+    Iso15693Parser* parser;
+} FuriHalNfcIso15693Listener;
+
+typedef struct {
+    // 4 bits per data bit on transmit
+    uint8_t fifo_buf[FURI_HAL_NFC_ISO15693_POLLER_MAX_BUFFER_SIZE * 4];
+    size_t fifo_buf_bits;
+    uint8_t frame_buf[FURI_HAL_NFC_ISO15693_POLLER_MAX_BUFFER_SIZE * 2];
+    size_t frame_buf_bits;
+} FuriHalNfcIso15693Poller;
+
+static FuriHalNfcIso15693Listener* furi_hal_nfc_iso15693_listener = NULL;
+static FuriHalNfcIso15693Poller* furi_hal_nfc_iso15693_poller = NULL;
+
+static FuriHalNfcIso15693Listener* furi_hal_nfc_iso15693_listener_alloc() {
+    FuriHalNfcIso15693Listener* instance = malloc(sizeof(FuriHalNfcIso15693Listener));
+
+    instance->signal = iso15693_signal_alloc(&gpio_spi_r_mosi);
+    instance->parser =
+        iso15693_parser_alloc(&gpio_spi_r_miso, FURI_HAL_NFC_ISO15693_MAX_FRAME_SIZE);
```

Comment:
The use of MISO here is an issue. It breaks emulation against readers that use ASK modulation instead of OOK modulation. For example legacy iClass readers https://github.com/flipperdevices/flipperzero-good-faps/pull/62#issuecomment-1776169972

The previous code uses the IRQ line instead as per https://github.com/flipperdevices/flipperzero-firmware/blob/35c903494c511f41a518dd045be5ab8991fa3991/lib/nfc/protocols/nfcv.c#L1223 which allows this to work.

Note that due to the diode on the IRQ line, using that pin requires enabling a PullDown which the old code achieved here: https://github.com/flipperdevices/flipperzero-firmware/blob/35c903494c511f41a518dd045be5ab8991fa3991/lib/nfc/protocols/nfcv.c#L1229

@pcunning